### DISCUSSION TITLE
Exporting again with --update will update some existing "_edited" files.

### DISCUSSION THREAD

**Discussion by chengguangnan** [RhetTbull/osxphotos#1385] [Q&A]
Hello, I'm trying to understand a behavor. 

This is the output from a second export of an album with `--photokit`:

```
Exporting 1840 photos ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Processed: 1840 photos, exported: 3, updated: 72, skipped: 1832, updated EXIF data: 0, missing: 18, error: 0
Elapsed time: 2:29:48
```

I found that export again with `--update` will update 72 files and those updated files have "_edited" in filename. BTW  `ls -1 | ag _edited | wc` shows that there are 85 "_edited" files, so not all of them were updated but a majority of them were. 

Those files were exported succesfully at the first export, so I don't know why they got updated again during the second export. 

Any idea what's the reason behind this? `shasum` sugguests that the updated file are idential to previous export. I was assuming maybe some kind of Finder meta attribute was updated?

**Comment by RhetTbull** [comment]
Not sure why, but if you run `osxphotos exportdb /path/to/export --history /path/to/image_edited.jpeg`  replacing the paths as appropriate, osxphotos will print out the history of why those photos were exported and that will give us some clues. For example, it will look something like this:

```
2024-01-13 16:16:10.139, new, None
2024-01-13 16:16:19.301, skip, None
2024-01-13 16:16:48.284, skip, None
2024-01-15 13:04:28.815, update: EDITED_SIG_DIFFERENT, None
```

`new` indicates the first export (photo was new), it was skipped twice, then it was updated because the signatures were different from what was stored in the database.

**Comment by chengguangnan** [comment]
It seems that the cause is that, my photo libaray has many videos with the same original filename, eg IMG_0001.mov (2019) and IMG_0001.mov (2022, new phone). I didn't verify this, but it seems that upon exporting, those videos were overwriting each other and cause some EDITED_SIG_DIFFERENT?

I added --filename "{original_name}_{id:06d}" and now exported files are skipped very fast.

**Reply by RhetTbull** [reply]
> those videos were overwriting each other 

Are you using `--overwrite`? If not then osxphotos should never overwrite a file with a different photo. osxphotos maintains a mapping of UUID to filename and if two filenames conflicts, increments them: `IMG_0001.mov`, `IMG_0001 (1).mov`, `IMG_0001 (2).mov`, etc.  Unless you use `--overwrite` which isn't recommended for maintaining a backup with `--update`.

**Reply by chengguangnan** [reply]
I was not using `--overwrite`, and "updated files" seems have content unchanged, maybe just a `mtime` touch?
